### PR TITLE
Aj 641 versions

### PR DESCRIPTION
**Pull request by calypsomatic**
This PR creates a new tag upon push to main first, and then, using the new tag, publishes all the clients and images to ensure they are in sync.  Docker images are given both the git hash and the tag.

### CODE REVIEW

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -33,22 +40,26 @@ jobs:
           echo "::set-output name=git_short_sha::${git_short_sha}"
       - name: Construct GCR docker image name and tag
         id: gcr-image-name
-        run: echo ::set-output name=name::${GCR_REGISTRY}/${GOOGLE_PROJECT}/${SERVICE_NAME}:${{ steps.setHash.outputs.git_short_sha }}
+        run: echo ::set-output name=name::${GCR_REGISTRY}/${GOOGLE_PROJECT}/${SERVICE_NAME}
```
Comment (aaronkanzer) [inline .github/workflows/publish-docker.yml:43]: Should we be referencing the style prescribed here as it seems `::set-output` will soon be deprecated: 

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples

File: .github/workflows/publish-docker.yml [hunk 2]
```diff
@@ -64,17 +75,20 @@ jobs:
...
+      - name: Add version tag to ACR
+        run: docker image tag ${{ steps.acr-image-name.outputs.name }}:${{ steps.setHash.outputs.git_short_sha }} ${{ steps.acr-image-name.outputs.name }}:${{ inputs.new-tag }}
 
       - name: Push image to Azure
         run: |
           echo ${{ secrets.ACR_SP_PASSWORD }} | docker login ${ACR_REGISTRY} \
           --username ${{ secrets.ACR_SP_USER }} \
           --password-stdin
-          docker push ${{ steps.acr-image-name.outputs.name }}
+          docker push --all-tags ${{ steps.acr-image-name.outputs.name }}
```
Comment (aaronkanzer) [inline .github/workflows/publish-docker.yml:91]: What is the reason to push multiple tags to the registry?

File: .github/workflows/publish-docker.yml [hunk 3]
```diff
@@ -33,22 +40,26 @@ jobs:
...
         run: |
           ./gradlew --build-cache jibDockerBuild \
-          --image=${{ steps.gcr-image-name.outputs.name }} \
+          --image=${{ steps.gcr-image-name.outputs.name }}:${{ steps.setHash.outputs.git_short_sha }} \
           -Djib.console=plain
+          
+      - name: Add version tag to GCR
+        run: docker image tag ${{ steps.gcr-image-name.outputs.name }}:${{ steps.setHash.outputs.git_short_sha }} ${{ steps.gcr-image-name.outputs.name }}:${{ inputs.new-tag }}
+          
       - name: Run Trivy vulnerability scanner
```
Comment (aaronkanzer) [inline .github/workflows/publish-docker.yml:58]: Just out of curiousity, why do we not run Trivy on images pushed to ACR?

File: .github/workflows/publish-docker.yml [hunk 4]
```diff
@@ -33,22 +40,26 @@ jobs:
           echo "::set-output name=git_short_sha::${git_short_sha}"
       - name: Construct GCR docker image name and tag
         id: gcr-image-name
-        run: echo ::set-output name=name::${GCR_REGISTRY}/${GOOGLE_PROJECT}/${SERVICE_NAME}:${{ steps.setHash.outputs.git_short_sha }}
+        run: echo ::set-output name=name::${GCR_REGISTRY}/${GOOGLE_PROJECT}/${SERVICE_NAME}
 
       - name: Construct ACR docker image name and tag
```
Comment (davidangb) [inline .github/workflows/publish-docker.yml:45]: same here, remove "and tag"?

**Review by davidangb — COMMENTED** [review]
I like this setup! I also don't have ACR access yet so can't verify what's there, let's remember to do that once we get access.

How about making a manual change to `build.gradle` to set `version = 0.2.0-SNAPSHOT` in this PR, to do a soft reset of our versioning?

File: .github/workflows/publish-docker.yml (hunk 2 above)
Comment (davidangb) [reply .github/workflows/publish-docker.yml:91]: some systems like our helm charts use the short git hash, but we'd like that hash to be associated with the semantic version number we're using for our clients and github tags. Having both tags on the image makes that nice and easy:
Screenshot 10-26-2022 at 05 02 PM

File: .github/workflows/publish-docker.yml (hunk 3 above)
Comment (calypsomatic) [reply .github/workflows/publish-docker.yml:58]: I guess I was assuming it's the exact same image, just with a different name/tag

**Review by aaronkanzer — APPROVED** [review]

**Review by sagehen03 — APPROVED** [review]

**Pull request merged**